build.gradle

```diff
@@ -50,64 +47,83 @@ minecraft {
         args << '-Dmixin.checks.interfaces=true'
         args << '-Dmixin.debug.export=true'
     }
-    clientJvmArgs.addAll(args)
-    serverJvmArgs.addAll(args)
+    extraRunJvmArguments.addAll(args)
+
+    useDependencyAccessTransformers = true
+
+    injectedTags.put("VERSION", project.version)
+}
+
+// Generate a my.project.Tags class with the version number as a field
+tasks.injectTags.configure {
+    outputClassName.set("${project.group}.Tags")
 }
 
 repositories {
     maven {
         url = 'https://maven.cleanroommc.com'
     }
+    maven { url = "https://repo.spongepowered.org/maven" }
+    //maven { url "https://maven.mcmoddev.com/" }
     maven {
-        url = 'https://repo.spongepowered.org/maven'
-    }
-    maven {
-        url = "https://www.cursemaven.com"
+        url "https://cursemaven.com"
+        content {
+            includeGroup "curse.maven"
+        }
     }
+    mavenLocal() // Must be last for caching to work
 }
 
 dependencies {
-    implementation 'org.jetbrains:annotations:20.1.0'
     if (project.use_assetmover.toBoolean()) {
-        deobfCompile 'com.cleanroommc:assetmover:0.2'
+        implementation 'com.cleanroommc:assetmover:2.0'
     }
     if (project.use_mixins.toBoolean()) {
-        deobfCompile 'zone.rong:mixinbooter:5.0'
+        implementation 'zone.rong:mixinbooter:7.0'
     }
 
-    // format for dependencies: deobfCompile 'curse.maven:<project-slug>-<project-id>:<file-id>'
-
-    // CCL 3.2.3.358
-    deobfCompile 'curse.maven:codechicken-lib-1-8-242818:2779848'
-    // GTCEu 2.6.0/2.6.1
-    //deobfCompile 'curse.maven:gregtech-ce-unofficial-557242:4480237'
-    deobfCompile 'curse.maven:gregtech-ce-unofficial-557242:4483462'
-
-    // GCYM 1.2.0
-    //deobfCompile 'curse.maven:gregicality-multiblocks-564858:4422413'
-    deobfCompile 'curse.maven:gregicality-multiblocks-564858:4480254'
-
-    //GeckoLib
-    deobfCompile 'curse.maven:geckolib-388172:4020277'
+    if (project.use_mixins.toBoolean()) {
+        api ("org.spongepowered:mixin:0.8.3") {transitive = false}
+        annotationProcessor('org.ow2.asm:asm-debug-all:5.2')
+        annotationProcessor('com.google.guava:guava:24.1.1-jre')
+        annotationProcessor('com.google.code.gson:gson:2.8.6')
+        annotationProcessor("org.spongepowered:mixin:0.8.3") {transitive = false}
+    }
 
-    // JEI 4.16.1.302
-    deobfCompile 'curse.maven:jei-238222:3043174'
-    // 1.4.28
-    deobfCompile 'curse.maven:the-one-probe-245211:2667280'
+    implementation 'org.jetbrains:annotations:24.0.1'
```

Comment:
This should use `compileOnly`. Right now forge tries to load it as a mod during game loading.